### DISCUSSION TITLE
Add a Particle Emitter Editor (or at least a Preview)

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3645] [Enhancement ideas (and their technical discussions)]
## Description

GDevelop allows users to select settings for particle emitters, but the user must add the object to a game and preview it before they can see what their changes look like.  This process could be greatly accelerated if users could immediately see what the particle emitter looks like as changes are made to settings. 

### Simple option
- Add a preview window that shows the current look of the emitter
- Preferably this is always visible and active in real-time
- If simpler to implement, add a "Preview Emitter" button that opens a preview that only includes the particle emitter. 

### Advanced option
- Build a tool that lets users edit particle emitter settings and 
- For example:
- https://pixijs.io/pixi-particles-editor/#pixieDust
- https://github.com/koreezgames/phaser3-particle-editor
- https://github.com/libgdx/libgdx/wiki/2D-Particle-Editor
- https://docs.godotengine.org/en/stable/tutorials/2d/particle_systems_2d.html

## Related

-[ Ancient forum thread about adding the Pixi particle editor: ](https://forum.gdevelop-app.com/t/demo-javascript-pixi-particles/11243/4)

**Comment by 4ian** [comment]
I see the idea! I was figuring the "live preview" feature ("apply changes to preview") could be used for this. 
It's not as convenient as a real time editor though (it's only when you click on it, so "semi real time"), but could still be used?

**Comment by Keevle** [comment]
> I see the idea! I was figuring the "live preview" feature ("apply changes to preview") could be used for this.
> It's not as convenient as a real time editor though (it's only when you click on it, so "semi real time"), but could still be used?

That's exactly what tristan wants a solution to tho, it's very clunky to have to edit a variable, then preview to see a change then repeat that process for every variable. @4ian

**Comment by 4ian** [comment]
Sure but I'm not sure Tristan actually used the Live Preview feature. So this can already cut in half the time for checking your changes, because you have a single button to click, with the editor staying open, and the game updating without restarting.
Then we can build on top of this and do something like automatically applying the changes. 

I'm not actually sure many people are even aware of this feature.

**Comment by tristanbob** [comment]
> Sure but I'm not sure Tristan actually used the Live Preview feature. So this can already cut in half the time for checking your changes, because you have a single button to click, with the editor staying open, and the game updating without restarting.
> Then we can build on top of this and do something like automatically applying the changes.
> 
> I'm not actually sure many people are even aware of this feature.

I do use the Live Preview feature, and it is helpful for testing particle emitters.  (BTW, I can't renew my subscription because my bank always declines the purchase.  I can still use LivePreview, just with seeing the reminder).

What is the feasibility of adding the Pixi particle editor as an external component, like Pixel and JFXR?   Users could create and modify the emitters with that tool, and when they hit save, it would be ready for use in GDevelop.

**Comment by arthuro555** [comment]
> (I can still use LivePreview, just with seeing the reminder)

That's because the live preview feature isn't paid, the popup is just a bit misleading (see #2544)
